manim/animation/composition.py

```diff
@@ -134,37 +136,59 @@ def init_run_time(self, run_time) -> float:
             The duration of the animation in seconds.
         """
         self.build_animations_with_timings()
-        if self.anims_with_timings:
-            self.max_end_time = np.max([awt[2] for awt in self.anims_with_timings])
-        else:
-            self.max_end_time = 0
+        # Note: if lag_ratio < 1, then not necessarily the final animation's
+        # end time will be the max end time!
```

Comment:
Please elaborate.

Reply:
Let's say you play a 10 seconds animation and then a 1 second animation, with a lag ratio of 0.1.
The 1° animation starts at t = 0.
At t = 1 the 2° one will begin, and will end at t = 2.
However, the 1° animation is still playing, and will end at t = 10.
So, the final animation's end time (t = 2) is not the AnimationGroup's end time (the greatest one, t = 10).

This doesn't happen with a lag ratio >= 1, because the next animation always plays after the previous one has finished, so the final animation's end time will always be the group's end time.

Reply:
Ah, I misunderstood the comment: I thought of the "final" animation to be the one that is playing the longest, not the final element in the list, and then got confused. I like the new, more explicit version  much better already, I'll read it more carefully after I've had some coffee. Thanks!